SHMEM/isx.c

```diff
@@ -722,106 +731,62 @@ static void print_run_info(FILE * fp)
 
 /*
  * Prints all of the timining information for an individual PE as a row
- * to the file specificed by 'fp'. 
+ * to the file specificed by 'fp'.
  */
 static void print_timer_values(FILE * fp)
 {
-  unsigned int num_records = NUM_PES * NUM_ITERATIONS; 
-
-  for(uint64_t i = 0; i < num_records; ++i) {
+  for(uint64_t i = 0; i < NUM_ITERATIONS; ++i) {
     for(int t = 0; t < TIMER_NTIMERS; ++t){
-      if(timers[t].all_times != NULL){
-        fprintf(fp,"%f\t", timers[t].all_times[i]);
+      if(timers[t].seconds_iter > 0){
+        fprintf(fp,"%f\t", timers[t].seconds[i]);
       }
-      if(timers[t].all_counts != NULL){
-        fprintf(fp,"%u\t", timers[t].all_counts[i]);
+      if(timers[t].count_iter > 0){
+        fprintf(fp,"%u\t", timers[t].count[i]);
       }
     }
     fprintf(fp,"\n");
   }
 }
 
-/* 
+/*
  * Aggregates the per PE timing information
- */ 
+ */
 static double * gather_rank_times(_timer_t * const timer)
 {
   if(timer->seconds_iter > 0) {
 
     assert(timer->seconds_iter == timer->num_iters);
```

Comment:
With the changes in this PR, you're exploiting the fact each PE has the same number of iterations, so the average of the averages across all PEs is equal to the average of the entire collection (not true if PEs have different # of iterations).  We _could_ verify that num_iters is constant across PEs with a gather or something... but perhaps a comment is sufficient explaining what's happening?